packages/std/Cargo.toml

```diff
@@ -36,17 +36,19 @@ ibc3 = ["stargate"]
 # This feature makes `BankQuery::Supply` available for the contract to call, but requires
 # the host blockchain to run CosmWasm `1.1.0` or higher.
 cosmwasm_1_1 = []
+# enable no_std support
+no_std = []
 
 [dependencies]
 base64 = "0.13.0"
 cosmwasm-derive = { path = "../derive", version = "1.1.5" }
 derivative = "2"
 forward_ref = "1"
 hex = "0.4"
-schemars = "0.8.3"
+schemars = { version = "0.8.3" }
 serde = { version = "1.0.103", default-features = false, features = ["derive", "alloc"] }
 serde-json-wasm = { version = "0.4.1" }
```

Comment:
How can this work without https://github.com/CosmWasm/serde-json-wasm/issues/40 being implemented first?

Reply:
This definitely is not going to compile in `no_std` and tbh I'd move all the types to a new crate like `std-types` with almost 0 deps that is `no_std` compatible by design. Then reexport to keep backward transparent compatibility, what do you think @webmaster128? Because right now, there is deps like `forward_ref` which is not `no_std` etc... I'd say extracting the types is the easiet thing to do.

Reply:
Hmm, everything is a type and whenever I hear "types", "utils", "helpers" or "models", it smells like code pushed to some corner instead of separation by domain that causes maintainability issues sooner or later.

But I guess what you are looking for is the interface layer over which the contract and the VM communicate. The types seralized on one side and deserialized on the other side.

Large parts of cosmwasm-std are contract only and don't need to live in this layer. Looking at the usages of cosmwasm_std inside of packages/vm/src, this is probably very minimal. I think this is a valid approach if we find a good separation. 

But if I understand correctly, this would be a major strategy shift compared to this PR. Then cosmwasm-std would not need to be no_std.